From 8c37b89ff43a771cf4d7043b8a199a279f39eb7e Mon Sep 17 00:00:00 2001
From: Palmer Cox <p@lmercox.com>
Date: Sat, 22 Feb 2025 23:00:57 -0500
Subject: [PATCH] openscenegraph: Fixing compiling on Darwin

The pull request on OpenSceneGraph contains an explanation of why this
is necessary:
https://github.com/openscenegraph/OpenSceneGraph/pull/1360. This fixes
compiling on Darwin, which was broken by the change to have libtiff
compiled with CMake in 8b724434c37a12ea5ed23ba55339e3dc72d86d7d which
was merged as part of https://github.com/NixOS/nixpkgs/pull/366566.
---
 pkgs/development/libraries/openscenegraph/default.nix | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/pkgs/development/libraries/openscenegraph/default.nix b/pkgs/development/libraries/openscenegraph/default.nix
index ee13f76d7feedf..c28beee8645c5f 100644
--- a/pkgs/development/libraries/openscenegraph/default.nix
+++ b/pkgs/development/libraries/openscenegraph/default.nix
@@ -137,6 +137,11 @@ stdenv.mkDerivation rec {
       url = "https://gitweb.gentoo.org/repo/gentoo.git/plain/dev-games/openscenegraph/files/openscenegraph-3.6.5-openexr3.patch?id=0f642d8f09b589166f0e0c0fc84df7673990bf3f";
       hash = "sha256-fdNbkg6Vp7DeDBTe5Zso8qJ5v9uPSXHpQ5XlGkvputk=";
     })
+    # Fix compiling with libtiff when libtiff is compiled using CMake
+    (fetchurl {
+      url = "https://github.com/openscenegraph/OpenSceneGraph/commit/9da8d428f6666427c167b951b03edd21708e1f43.patch";
+      hash = "sha256-YGG/DIHU1f6StbeerZoZrNDm348wYB3ydmVIIGTM7fU=";
+    })
   ];
 
   cmakeFlags =
